x/mongo/driver/session/server_session.go

```diff
@@ -27,12 +28,18 @@ type Server struct {
 
 // returns whether or not a session has expired given a timeout in minutes
 // a session is considered expired if it has less than 1 minute left before becoming stale
-func (ss *Server) expired(timeoutMinutes uint32) bool {
-	if timeoutMinutes <= 0 {
+func (ss *Server) expired(topoDesc topologyDescription) bool {
+	// There is no server monitoring in LB mode, so we do not track session timeout minutes from server hello responses
```

Comment:
To double check, The connection's server description should still have `SessionTimeoutMinutes` set from the handshake. But because monitoring does not occur, `topoDesc.SessionTimeoutMinutes` should remain 0, right?

Are there are checks for session support based on the topology description's `SessionTimeoutMinutes`? The session spec says:

> A driver can determine whether a deployment supports sessions by checking whether the logicalSessionTimeoutMinutes property of the TopologyDescription has a value or not. 

But `Operation.addSession` only checks the server description. So I think this is OK.

Reply:
There shouldn't be any checks based on the topology description. All checks in `Operation` should be based on the connection description, which will have `logicalSessionTimeoutMinutes` set. There are also incoming integration tests for session support in LB mode, which do pass with this change.